Wrappers/Python/ccpi/optimisation/functions/MixedL21Norm.py

```diff
@@ -136,6 +95,8 @@ def proximal(self, x, tau, out=None):
             tmp = x.pnorm(2)
             res = (tmp - tau).maximum(0.0) * x/tmp
             
+            # TODO avoid using numpy, add operation in the framework
+            # This will be useful when we add cupy 
             for el in res.containers:
```

Comment:
This will not work if the `DataContainer` is not wrapping a numpy array, i.e. in the case of SIRF `DataContainer`s.
I suppose we should use
```python
elarray = el.as_array()
elarray[np.isnan(elarray)]=0
el.fill(elarray)
```
Then `fill` should realise that if the array we are using to fill our datacontainer is the same as the wrapped one it should do nothing. In this case we would maintain the ability to use this function with SIRF objects and other containers.

Reply:
I do not think that this will work. It is a BlockDataContainer andneed to access them and check if they have nan values. I thought that we did a BlockDataContainer version in SIRF, no?

Reply:
`BlockDataContainer` contain a `list` of objects. If you pass a SIRF or CIL `DataContainer` it's totally equal as they present the same interface.
 
You say that it will not work, what? If `el` is a SIRF or CIL `DataContainer` `elarray` will be a `numpy.ndarray`. Both CIL and SIRF `DataContainer`s have the `fill` method, so that'll work.